pkg/server/plugin/keymanager/awskms/awskms_test.go

```diff
@@ -0,0 +1,1370 @@
+					AliasLastUpdatedDate: &unixEpoch,
+				},
+			},
+
+			expectedEntries: []fakeKeyEntry{
+				{
+					AliasName: aws.String("alias/test_example_org/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/id_01"),
+				},
+				{
+					AliasName: aws.String("alias/test_example_org/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/id_02")},
+				{
+					AliasName: aws.String("alias/another_td/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/id_04"),
+				},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			// setup
+			ts := setupTest(t)
+			ts.fakeClient.setEntries(tt.fakeEntries)
+			ts.plugin.hooks.deleteDone = make(chan struct{}, 3)
+
+			// exercise
+			_, err := ts.plugin.Configure(ctx, tt.configureRequest)
+			require.NoError(t, err)
+			// TODO: it won't work without it
+			runtime.Gosched()
+			ts.clockHook.Add(48 * time.Hour)
+			<-ts.plugin.hooks.deleteDone
+
+			// assert
+			storedAliases := ts.fakeClient.store.aliases
+			require.Len(t, storedAliases, len(tt.expectedEntries))
+			for _, expected := range tt.expectedEntries {
+				_, ok := storedAliases[*expected.AliasName]
+				require.True(t, ok, "Expected alias was not present on end result: %q", *expected.AliasName)
+			}
+		})
+	}
+}
+
+func TestDisposeKeys(t *testing.T) {
+	for _, tt := range []struct {
+		name             string
+		configureRequest *plugin.ConfigureRequest
+		fakeEntries      []fakeKeyEntry
+		expectedEntries  []fakeKeyEntry
+	}{
+		{
+			name:             "dispose keys succeeds",
+			configureRequest: configureRequestWithDefaults(t),
+
+			fakeEntries: []fakeKeyEntry{
+				{
+					AliasName:            aws.String("alias/test_example_org/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/id_01"),
+					KeyID:                aws.String("key_id_01"),
+					Description:          aws.String("SPIRE_SERVER_KEY/test_example_org"),
+					KeySpec:              types.CustomerMasterKeySpecRsa4096,
+					Enabled:              true,
+					PublicKey:            []byte("foo"),
+					CreationDate:         &unixEpoch,
+					AliasLastUpdatedDate: &unixEpoch,
+				},
+				{
+					AliasName:            nil,
+					KeyID:                aws.String("key_id_02"),
+					Description:          aws.String("SPIRE_SERVER_KEY/test_example_org"),
+					KeySpec:              types.CustomerMasterKeySpecRsa2048,
+					Enabled:              true,
+					PublicKey:            []byte("foo"),
+					CreationDate:         &unixEpoch,
+					AliasLastUpdatedDate: &unixEpoch,
+				},
+				{
+					AliasName:            aws.String("alias/not_spire/id_03"),
+					KeyID:                aws.String("key_id_03"),
+					Description:          aws.String("SPIRE_SERVER_KEY/test_example_org"),
+					KeySpec:              types.CustomerMasterKeySpecEccNistP384,
+					Enabled:              true,
+					PublicKey:            []byte("foo"),
+					CreationDate:         &unixEpoch,
+					AliasLastUpdatedDate: &unixEpoch,
+				},
+				{
+					AliasName:            aws.String("alias/another_td/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/id_04"),
```

Comment:
What exactly are we matching on when we dispose of keys? Disposing of a KMS key is of course a very sensitive and error-prone operation, we want to be extra sure that we're only deleting the ones we created and manage. I think we should add some test cases here, and perhaps some more verification in the plugin code.

I may have misread, but it seems like we're only looking for trust domain name in the alias... and maybe a static `alias/` prefix. Trust domain name is often something with meaning beyond just SPIFFE/SPIRE ... for example, the trust domain name may be the name of the company e.g. `acme.co`. In that case, it's easy to imagine that we're not too far off from a name collision and possibly deleting keys that aren't ours

Reply:
That's a great observation. Aliases now look like this: `alias/SPIRE_SERVER/{TRUST_DOMAIN}/{SERVER_ID}/{KEY_ID}`. The prefix 'SPIRE_SERVER' was added with the intention of making collisions less likely.